### PR TITLE
Fix minor issues with the chaining functionality

### PR DESCRIPTION
**Pull request by shardulc**

### CODE REVIEW

File: assets/js/translator.js [hunk 1]
```diff
@@ -730,7 +731,7 @@ function detectLanguage() {
 }
 
 function translationNotAvailable() {
-    $('#translatedText').text(dynamicLocalizations['Not_Available']);
+    $('#translatedText').val(dynamicLocalizations['Not_Available']);
```
Comment (sushain97) [inline assets/js/translator.js]: Can we actually just change both val and text? I'm not sure of the intricacies behind this and it won't hurt.

File: assets/js/translator.js [hunk 2]
```diff
@@ -20,7 +20,8 @@ if(modeEnabled('translation')) {
 
         function getDstLangs(srcLang) {
             var targets = [];
-            var targetsSeen = {srcLang: true};
+            var targetsSeen = {};
+            targetsSeen[srcLang] = true;
```
Comment (sushain97) [inline assets/js/translator.js]: Why did this become two lines? It looks the same??

File: assets/js/translator.js [hunk 3]
```diff
@@ -572,7 +573,7 @@ function inputFile() {
 }
 
 function translateDoc() {
-    var validPair = pairs[curSrcLang] && pairs[curSrcLang].indexOf(curDstLang) !== -1,
+    var validPair = originalPairs[curSrcLang] && originalPairs[curSrcLang].indexOf(curDstLang) !== -1,
```
Comment (sushain97) [inline assets/js/translator.js]: This doesn't change the list, though?

Comment (sushain97) [inline assets/js/translator.js]: This doesn't change the list, though?

File: assets/js/translator.js (hunk 2 above)
Comment (shardulc) [inline assets/js/translator.js]: Seems to work differently: the old one adds a key called 'srcLang', while the new one actually expands the variable.

File: assets/js/translator.js (hunk 3 above)
Comment (shardulc) [inline assets/js/translator.js]: We want to preserve the checked/unchecked-ness of the checkbox, right?

File: assets/js/translator.js (hunk 2 above)
Comment (sushain97) [inline assets/js/translator.js]: OH, right. I've definitely made this same mistake before. Easy bug to miss :)

File: assets/js/translator.js (hunk 3 above)
Comment (sushain97) [inline assets/js/translator.js]: Yep.

**Pull request merged**